pkg/operator/ceph/object/admin.go

```diff
@@ -323,7 +323,19 @@ func GetAdminOPSUserCredentials(objContext *Context, spec *cephv1.ObjectStoreSpe
 		AdminOpsUser: true,
 	}
 	logger.Debugf("creating s3 user object %q for object store %q", userConfig.UserID, ns)
-	user, rgwerr, err := CreateUser(objContext, userConfig)
+
+	forceUserCreation := false
+	// If the cluster where we are running the rgw user create for the admin ops user is configured
+	// as a secondary cluster the gateway will error out with:
+	// 		Please run the command on master zone. Performing this operation on non-master zone leads to
+	// 		inconsistent metadata between zones
+	// It is safe to force it since the creation will return that the user already exists since it
+	// has been created by the primary cluster. In this case, we simply read the user details.
+	if spec.IsMultisite() {
+		forceUserCreation = true
+	}
+
+	user, rgwerr, err := CreateUser(objContext, userConfig, forceUserCreation)
```

Comment:
@leseb : Just wondering if we call `GetUser()` before the `CreateUser()`, will it resolve this issue than forcing the user creation and fetching already existing user information from `CreateUser()`

Reply:
That works too.

Reply:
IMHO that is a bit cleaner approach

Reply:
Yes, I mixed up with an old bug in Ceph which was related to the Admin Ops API. However, this is not using the admin ops but the CLI directly. Agreed it's cleaner. Will fix.